lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -82,80 +130,94 @@ def is_active(self) -> bool:
             logger.error("Error while checking Vault health status: %s", e)
             return False
 
-    def is_api_available(self) -> bool:
-        """Return whether Vault is available."""
-        try:
-            self._client.sys.read_health_status(standby_ok=True)
-            return True
-        except (VaultError, RequestException) as e:
-            logger.error("Error while checking Vault health status: %s", e)
-            return False
-
-    def unseal(self, unseal_keys: List[str]) -> None:
-        """Unseal Vault."""
-        for unseal_key in unseal_keys:
-            self._client.sys.submit_unseal_key(unseal_key)
-        logger.info("Vault is unsealed")
+    def enable_audit_device(
+        self, device_type: Literal["file", "syslog", "socket"], path: str
+    ) -> None:
+        """Enable a new audit device at the supplied path.
 
-    def set_token(self, token: str) -> None:
-        """Set the Vault token for authentication."""
-        self._client.token = token
+        Args:
+            device_type: One of three available device types
+            path: The path that will receive audit logs
+        """
+        try:
+            self._client.sys.enable_audit_device(
+                device_type=device_type,
+                options={"file_path": path},
+            )
+            logger.info("Enabled audit device %s for path %s", device_type, path)
+        except InvalidRequest:
+            logger.info("Audit device already enabled.")
 
-    def approle_login(self, role_id: str, secret_id: str) -> None:
-        """Authenticate to Vault using AppRole."""
-        self._client.auth.approle.login(role_id=role_id, secret_id=secret_id, use_token=True)
+    def enable_auth_method(self, auth_method: Literal["approle"]) -> None:
```

Comment:
I am not convinced that having a parameter that takes a single value is a great API. Do we have other auth methods planned for support in the near future? Otherwise I would keep it simple and make this method only enable approle.

Reply:
I liked the readability and the consistency with other methods of `enable_auth_method('approle')` more than `enable_approle_auth_method()`, but I'm not opposed to changing it. I don't think either charm will ever use anything other than approles, although there may be a case where a user may want to enable it through the charm.
Although that's very far away in the future.

Reply:
Changed.